### PR TITLE
Cosmos upgrade to v0.43.0-rc2.agoric

### PR DESCRIPTION
**Pull request by michaelfig**
Take most recent cosmos-sdk with our changes.  These ones include @zmanian and @JimLarson changes to vesting, as well as the most recent v0.43.0-rc2 changes from cosmos-sdk.  We want eventually to be on the v0.43.x official release when it is available.

Also, two minor cleanups in ag-solo.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -31,7 +31,7 @@ replace google.golang.org/grpc => google.golang.org/grpc v1.33.2
 
 // At least until GetABCIEventHistory() is implemented and released.
 // And also `gentx --keyring-dir=...`
-replace github.com/cosmos/cosmos-sdk => github.com/agoric-labs/cosmos-sdk v0.43.0-rc0.agoric
+replace github.com/cosmos/cosmos-sdk => github.com/agoric-labs/cosmos-sdk v0.43.0-rc2.agoric
```
Comment (dckc) [inline go.mod:34]: why are we upgrading? I see

> Take most recent cosmos-sdk with our changes

Which changes?

Where can I see the difference between rc0 and rc2? I don't see rc2 in https://github.com/agoric-labs/cosmos-sdk/releases

**Review by dckc — COMMENTED** [review]
I need more clues

File: go.mod (hunk 1 above)
Comment (michaelfig) [reply go.mod:34]: > Which changes?

https://github.com/agoric-labs/cosmos-sdk/tags

It's not a published release, just a tag.

Comment (dckc) [reply go.mod:34]: thans... found https://github.com/cosmos/cosmos-sdk/compare/v0.43.0-rc0...agoric-labs:v0.43.0-rc2.agoric ... not yet enlightened...

**Review by dckc — APPROVED** [review]
thanks for the additional context.

onward!

**Pull request merged**